### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by hanpeter**
Massive overhauling of `krux-boto`.

1. Linted every file per PEP8.
2. Change the `cli.py` to make use of new `krux.cli.Application` features.
3. Change the `Boto` constructor to take arguments rather than a parser.
4. Set the defaults of arguments to match CLI arguments.

### CODE REVIEW

**Comment by paul-krohn** [comment]
The pull request/changes themselves all look good to me. I do have one reservation about this module, and doing so much handling of credentials, in a way that repeats what boto does. 

My recommendation is to have the argument parser default to None for the credential values, and pass that to boto, which will then check for environment values and .boto configs.

@plathrop, @jib, any thoughts on the subject?

**Comment by plathrop** [comment]
I say +1, and let's discuss credentials elsewhere.

File: krux_boto/boto.py [hunk 1]
```diff
@@ -25,130 +25,152 @@
...
+NAME = 'krux-boto'
 
-### Constants
-ACCESS_KEY      = 'AWS_ACCESS_KEY_ID'
-SECRET_KEY      = 'AWS_SECRET_ACCESS_KEY'
-NAME            = 'krux-boto'
-DEFAULT_REGION  = 'us-east-1'
+# Defaults
+DEFAULT = {
+    'log_level': lambda: DEFAULT_LOG_LEVEL,
```
Comment (jib) [inline krux_boto/boto.py]: for the more novices pythonistas (i.e., me ;) - can you add a comment why 'lambda' and not just the value?

File: setup.py [hunk 2]
```diff
@@ -2,43 +2,45 @@
...
-    author_email     = 'jos@krux.com',
-    description      = 'Library for interacting with boto built on krux-stdlib',
-    url              = REPO_URL,
-    download_url     = DOWNLOAD_URL,
-    license          = 'All Rights Reserved.',
-    packages         = find_packages(),
-    install_requires = [
+    name='krux-boto',
+    version=VERSION,
+    author='Jos Boumans',
```
Comment (jib) [inline setup.py]: feel free to amend the authors list here... :)

**Comment by jib** [comment]
Looks like a clean refactor to me, +1 

agreed we can take the credentials conversation out of band.

**Pull request merged**